File: snippets/fsharp/VS_Snippets_CLR_System/system.Func~1/fs/Lambda.fs
module Lambda

// <Snippet4>
open System
open System.IO

type OutputTarget() =
    member _.SendToFile() =
        try
            let fn = Path.GetTempFileName()
            use sw = new StreamWriter(fn)
            sw.WriteLine "Hello, World!"
            true
        with _ ->
            false

let output = OutputTarget()
let methodCall = Func<bool>(fun () -> output.SendToFile())
if methodCall.Invoke() then
    printfn "Success!"
else
    printfn "File write operation failed."
// </Snippet4>

File: snippets/csharp/System/FuncTResult/Overview/Lambda.cs
// <Snippet4>
using System;
using System.IO;

public class LambdaExpressionExample
{
    public static void Run()
    {
        LambdaOutputTarget output = new();
        Func<bool> methodCall = () => output.SendToFile();
        if (methodCall())
            Console.WriteLine("Success!");
        else
            Console.WriteLine("File write operation failed.");
    }
}

public class LambdaOutputTarget
{
    public bool SendToFile()
    {
        try
        {
            string fn = Path.GetTempFileName();
            StreamWriter sw = new(fn);
            sw.WriteLine("Hello, World!");
            sw.Close();
            return true;
        }
        catch
        {
            return false;
        }
    }
}
// </Snippet4>
